react/tic-tac-toe/src/app/screens/Game/index.js

```diff
@@ -1,20 +1,86 @@
-import React from 'react';
+import React, { Component } from 'react';
 
 import styles from './styles.module.scss';
 import Board from './components/Board';
+import HistoryList from './components/HistoryList';
+import { boardLines } from './constants';
 
-function Game () {
-  return (
-    <div className={styles.game}>
-      <div className={styles.gameBoard}>
-        <Board />
-      </div>
-      <div className={styles.gameInfo}>
-        <div>{/* status */}</div>
-        <ol>{/* TODO */}</ol>
+class Game extends Component {
+  state = {
+    history: [{
+      squares: Array(9).fill(null)
+    }],
+    stepNumber: 0,
+    player: 'X'
+  };
+
+  changeTurn = () => this.state.player === 'X' ? 'O' : 'X';
+
+  jumpTo = (step) => {
+    const player = step % 2 ? 'X' : '0';
+    const history = this.state.history.slice(0, step + 1);
+    this.setState({
+      stepNumber: step,
+      player,
+      history
+    });
+  }
+
+  handleClick = index => {
+    const history = this.state.history.slice(0, this.state.stepNumber + 1);
+    const current = history[this.state.stepNumber];
+    const squares = [...current.squares];
+    const hasWon = this.calculateWinner(squares);
+    const isFilled = squares[index];
+
+    if (hasWon || isFilled) {
+      return;
+    }
+
+    squares[index] = this.state.player;
+    return this.setState({
```

Comment:
You don't need a return statement here :eyes: You just want to set a new state.

```suggestion
    this.setState({
```